src/gluonts/torch/model/tide/estimator.py

```diff
@@ -0,0 +1,419 @@
+    decoder_output_dim
+        Output size of dense decoder (default: 8).
+    dropout_rate
+        Dropout regularization parameter (default: 0.3).
+    num_feat_dynamic_proj
+        Output size of feature projection layer (default: 4).
+    num_feat_dynamic_real
+        Number of dynamic real features in the data (default: 0).
+    num_feat_static_real
+        Number of static real features in the data (default: 0).
+    num_feat_static_cat
+        Number of static categorical features in the data (default: 0).
+    cardinality
+        Number of values of each categorical feature.
+        This must be set if ``num_feat_static_cat > 0`` (default: None).
+    embedding_dimension
+        Dimension of the embeddings for categorical features
+        (default: ``[min(16, (cat+1)//2) for cat in cardinality]``).
+    layer_norm
+        Enable layer normalization or not (default: False).
+    lr
+        Learning rate (default: ``1e-3``).
+    weight_decay
+        Weight decay regularization parameter (default: ``1e-8``).
+    distr_output
+        Distribution to use to evaluate observations and sample predictions
+        (default: StudentTOutput()).
+    scaling
+        Which scaling method to use to scale the target values (default: mean).
+    loss
+        Loss to be optimized during training
+        (default: ``NegativeLogLikelihood()``).
+    batch_size
+        The size of the batches to be used for training (default: 128).
+    num_batches_per_epoch
+        Number of batches to be processed in each training epoch
+        (default: 50).
+    trainer_kwargs
+        Additional arguments to provide to ``pl.Trainer`` for construction.
+    train_sampler
+        Controls the sampling of windows during training.
+    validation_sampler
+        Controls the sampling of windows during validation.
+    """
+
+    @validated()
+    def __init__(
+        self,
+        freq: str,
+        prediction_length: int,
+        context_length: Optional[int] = None,
+        feat_proj_hidden_dim: Optional[int] = None,
+        encoder_hidden_dim: Optional[int] = None,
+        decoder_hidden_dim: Optional[int] = None,
+        temporal_hidden_dim: Optional[int] = None,
+        distr_hidden_dim: Optional[int] = None,
+        num_layers_encoder: Optional[int] = None,
+        num_layers_decoder: Optional[int] = None,
+        decoder_output_dim: Optional[int] = None,
+        dropout_rate: float = 0.2,
+        num_feat_dynamic_proj: Optional[int] = None,
+        num_feat_dynamic_real: int = 0,
+        num_feat_static_real: int = 0,
+        num_feat_static_cat: int = 0,
+        cardinality: Optional[List[int]] = None,
+        embedding_dimension: Optional[List[int]] = None,
+        layer_norm: bool = False,
+        lr: float = 1e-3,
+        weight_decay: float = 1e-8,
+        scaling: Optional[str] = "mean",
+        distr_output: DistributionOutput = StudentTOutput(),
+        loss: DistributionLoss = NegativeLogLikelihood(),
```

Comment:
Just a heads-up: in light of #3093, this should be removed (I'm reviewing #3093 first, we can then follow up on the required changes)

Reply:
@lostella adapted the changes.